pkg/kyverno/common/common.go

```diff
@@ -978,3 +982,35 @@ func GetKindsFromPolicy(policy *v1.ClusterPolicy) map[string]struct{} {
 	}
 	return kindOnwhichPolicyIsApplied
 }
+
+//GetPatchedResourceFromPath - get patchedResource from given path
+func GetPatchedResourceFromPath(fs billy.Filesystem, path string, isGit bool, policyResourcePath string) (unstructured.Unstructured, error) {
+	var patchedResourceBytes []byte
+	var patchedResource unstructured.Unstructured
+	var err error
+	if isGit {
+		if len(path) > 0 {
+			filep, err := fs.Open(filepath.Join(policyResourcePath, path))
+			if err != nil {
+				fmt.Printf("Unable to open patchedResource file: %s. error: %s", path, err)
+			}
+			patchedResourceBytes, err = ioutil.ReadAll(filep)
+		}
+		if err != nil {
+			fmt.Printf("\n----------------------------------------------------------------------\nfailed to load patchedResource: %s. \nerror: %s\n----------------------------------------------------------------------\n", path, err)
+		}
+		patchedResource, err = GetPatchedResource(patchedResourceBytes)
```

Comment:
Whether its `Git` or `filesystem`, once you extract the `bytes`, the rest process is same. can we write something like this
```
if git {
  // extract logic for patchedResourceBytes
} else {
  // extract logic for patchedResourceBytes
}
// get patchedResource from patchedResourceBytes
```

Reply:
> 
> 
> Whether its `Git` or `filesystem`, once you extract the `bytes`, the rest process is same. can we write something like this
> 
> ```
> if git {
>   // extract logic for patchedResourceBytes
> } else {
>   // extract logic for patchedResourceBytes
> }
> // get patchedResource from patchedResourceBytes
> ```

Yes, your way is much better. And it also reduces multiple print statement into single. Therefore, no need to create separate function also.